src/api/MainSolver.cc

```diff
@@ -82,7 +82,9 @@ void MainSolver::push()
 bool MainSolver::pop()
 {
     if (frames.frameCount() > 1) {
-        if (config.produce_inter() > 0) {
+        //? TK: is this really related only to interpolants, or to proofs in general?
+        //? e.g. in Theory, there is: `bool keepPartitions() const { return config.produce_proof(); }`
+        if (config.produce_inter()) {
```

Comment:
This should use the new method `MainSolver::trackPartitions` that I suggest to create below.